apps/rest-api-app/src/main/java/bisq/rest_api/endpoints/ReportApi.java

```diff
@@ -86,16 +100,47 @@ public ReportDto getReport(@Parameter(description = "address from which we reque
     )
     @GET
     @Path("get-reports/{addresses}")
-    public List<ReportDto> getReports(@Parameter(description = "comma separated addresses from which we request the report") @PathParam("addresses") String addresses) {
-        List<CompletableFuture<Report>> futures = CollectionUtil.streamFromCsv(addresses)
-                .map(address -> networkService.requestReport(Address.fromFullAddress(address)))
+    public List<ReportDto> getReports(
+            @Parameter(description = "comma separated addresses from which we request the report")
+            @PathParam("addresses") String addresses) {
+
+        List<String> addressList;
+        try {
+            addressList = CollectionUtil.streamFromCsv(addresses).toList();
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to parse addresses from CSV input: " + addresses);
+        }
+
+
+        List<CompletableFuture<ReportDto>> futures = addressList.stream()
+                .map(this::fetchReportForAddress)
                 .toList();
+
+        CompletableFuture<Void> allFutures = CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
```

Comment:
We have a util method `CompletableFutureUtils.allOf` with an improved API (see comments in the method about the differences to CompletableFuture.allOf).